### PR TITLE
doc: MCUboot doc fixes

### PR DESCRIPTION
**Pull request by FrancescoSer**
Fixed note formatting.
Fixed typos in release notes.
Fixed case in chapter titles.
Removed contractions.
Fixed various spellings uncompliant with ncs doc guidelines.
Fixed references to flashing instead of programming.
Fixed commands formatting in Zephyr test plan.
Fixed references to doc folder in imgtool.md.
Removed mentions to Fabio Utzig in MCUboot sec policy as he left the company.
Connected to JIRA NCSDK 10376

Signed-off-by: Francesco Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

**Comment by FrancescoSer** [comment]
This is a test to see how an edit to the docs in the mcuboot repo would look like.
There is a wrapper folder in ncs that also wraps rst files inside the mcuboot docs. 
(for reference: \ncs\nrf\doc\mcuboot\wrapper.rst )

I'm wondering if that folder shouldn't be the preferred way to edit the mcuboot docs, instead of editing .md files.

File: docs/release.md
Comment (peknis) [inline docs/release.md:25]:
```suggestion
Before making a release, update the `docs/release-notes.md`
to describe the release. This should be a high-level description of
the changes, not a list of the git commits.
```

File: docs/release.md
Comment (peknis) [inline docs/release.md:29]:
```suggestion
Before each release, tags are made (see below) for at least one
```

File: docs/release.md
Comment (peknis) [inline docs/release.md:32]:
```suggestion
the official a.b.c release). The intent is to freeze the code for a
time, and allow testing.
```

File: docs/release.md
Comment (peknis) [inline docs/release.md:63]:
```suggestion
To make a release, make sure your local repo is on the tip version by
fetching from origin. Typically, the releaser should create a branch
```

File: docs/testplan-mynewt.md
Comment (peknis) [inline docs/testplan-mynewt.md:28]:
```suggestion
If testing RSA/PSS, `newt create-image` needs to be passed in the extra flag `--rsa-pss`, for example:
```

File: docs/testplan-zephyr.md
Comment (peknis) [inline docs/testplan-zephyr.md:4]:
```suggestion
The following roughly describes how MCUboot is tested on Zephyr. The
testing is performed with the code in `samples/zephyr`. These examples
```

File: docs/release-notes.md
Comment (peknis) [inline docs/release-notes.md:131]:
```suggestion
The 1.4.0 release of MCUboot primarily adds support for multi-image
booting. With this release, MCUboot can manage two images that can be
updated independently. It also supports additions to the
```

File: docs/release-notes.md
Comment (peknis) [inline docs/release-notes.md:93]:
```suggestion
This release of MCUboot works the Zephyr "main" at the time of the
release. It was tested as of hash `1a89ca1238`. When Zephyr v2.3.0 is
released, there will be a possible 1.6.1 or similar release of Zephyr
if needed to address any issues. There also may be branch releases of
```

File: docs/release-notes.md
Comment (peknis) [inline docs/release-notes.md:76]:
```suggestion
- CBOR decoding in serial recovery is replaced by code generated from a CDDL description.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by FrancescoSer** [comment]
@peknis Thank you for your feedback, however this was just a draft to test how the edits would look like in NCS.
Since it was later decided to contribute upstream, I opened this PR in the MCUboot repo instead:

https://github.com/mcu-tools/mcuboot/pull/1129

Also, it is still in draft. More edits are to come, but I'm waiting for some feedback from the MCUboot maintainers first.

**Pull request closed without merging**
